internal_ws/yktrace/src/hwt/mapper.rs

```diff
@@ -73,6 +75,7 @@ impl HWTMapper {
 
 /// Extract the program header offset. This offset can be used to translate the address of a trace
 /// block to a program address, allowing us to find the correct SIR location.
+/// XXX if this is an address should it be usize?
```

Comment:
We need to check this, but I think it's because `Elf_Addr` is `uint64_t` on x86_64. This code needs to be platform guarded.

Reply:
OK.

Reply:
Yes, so on a 64-bit platform, [ELF represents addresses as a `u64`](https://docs.rs/libc/0.2.86/libc/type.Elf64_Addr.html).

But we should define our own type (say `ElfAddr`) which is either a `u32` or a `u64` depending on platform. Isn't there a `target_pointer_width` `cfg` or something?

Reply:
[here](https://doc.rust-lang.org/reference/conditional-compilation.html#target_pointer_width)

Reply:
Looks like libipt is fixed to 64bit for now: https://github.com/intel/libipt/blob/905313ae4f066d0f79535ade8d4bc518bf49913c/libipt/include/intel-pt.h.in#L2231-L2233

Reply:
This is actually quite fiddly. I think it should be a separate PR.

Reply:
https://github.com/softdevteam/yk/issues/256